File: Erlang/lambdaconscarcdr.erl
Kons  = fun(Hd, Tl) ->
  fun
    (true ) -> Hd;
    (false) -> Tl
  end
end,

Virst = fun(L) -> L(true ) end,
Rrest = fun(L) -> L(false) end,

Lstt  = Kons(1, Kons(2, nil)),

Virst(Rrest(Lstt)).

File: Dart/lambdaconscarcdr.dart
#!/usr/bin/env dart

var kons  = (hd, [tl = null]) => (x) => x ? hd : tl;
var virst = (l) => l(true );
var rrest = (l) => l(false);

var lstt  = kons(1, kons(2));

virst(rrest(lstt));
